src/views/catalog/customize/components/CustomizeForms/CustomizeFormWithDisk.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { V1beta1DataVolumeSpec } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Form } from '@patternfly/react-core';
+
+import {
+  buildFields,
+  extractParameterNameFromMetadataName,
+  getTemplateStorageQuantity,
+} from '../../utils';
+import { ExpandableCustomizeSourceSection } from '../CustomizeSource/ExpandableCustomizeSourceSection';
+import { ExpandableOptionsFields } from '../ExpandableOptionalFields';
+import { FieldGroup } from '../FieldGroup';
+import { FormActionGroup } from '../FormActionGroup';
+
+import { useCustomizeFormSubmit } from './useCustomizeFormSubmit';
+
+type CustomizeFormWithDiskProps = {
+  template: V1Template;
+};
+
+export const CustomizeFormWithDisk: React.FC<CustomizeFormWithDiskProps> = ({ template }) => {
+  const { t } = useKubevirtTranslation();
+  const [customDiskSource, setCustomDiskSource] = React.useState<V1beta1DataVolumeSpec>();
+  const [onSubmit, loaded, error] = useCustomizeFormSubmit(template, customDiskSource);
+
+  const parameterForName = extractParameterNameFromMetadataName(template);
+
+  const [requiredFields, optionalFields] = buildFields(template, [parameterForName], t);
+
+  const onDiskSourceChange = React.useCallback((newDiskSource) => {
+    setCustomDiskSource(newDiskSource);
+  }, []);
+
+  return (
+    <Form onSubmit={onSubmit}>
+      {requiredFields?.map((field) => (
+        <FieldGroup key={field.name} field={field} showError={error} />
+      ))}
+
+      <ExpandableCustomizeSourceSection
+        onChange={onDiskSourceChange}
```

Comment:
you can't have a different onChange for the different sections, 
it's always
a - pass parameters
b - overide name
c - overide storage
d - add drivers disk
in one pass
there is no meaning to doing on change without first processing the parameters

Reply:
to clearify
you should have one onChange method
this method should
a - cakk a method that takes a template name+namespace + parameters and return a processed templates
b - take the vm object
c - update vm name and namespace
d - update main disk
e - add installCD if needed
f - add driversCD if needed
return the vm json ( do with it what needed for example create the VM / store it in some config map as json string / store it in session storage ... )

Reply:
okay, so the problem is that this process should not be in the onSubmit but in the onChange right?